halibot/halauth.py

```diff
@@ -1,11 +1,21 @@
 import json
 import logging
+from halibot.message import Message
 
 def hasPermission(perm, reply=False):
 	def real_dec(func):
-		def wrapper(self, msg, *args, **kwargs):
+		def wrapper(self, *args, **kwargs):
+			msg = None
+			for i in list(args) + list(kwargs.values()):
+				if i.__class__ == Message:
```

Comment:
What about subclasses of `Message`? I think we should be checking for those too.

Reply:
Yes, it should be checking if the class chain has `Message`.